### DISCUSSION TITLE
Manual loading state control as seen in Nuxt 2?

### DISCUSSION THREAD

**Discussion by Gaspadlo** [nuxt/nuxt#16351] [Questions] [answered]
Hello, I am trying to manually control nuxt's loading state.

Typically for example when submitting an ajax form you want to show the loading indicator consistently. This was possible in Nuxt 2 via `this.$nuxt.$loading['start' || 'finish']()`, but looking at the source code of the NuxtLoadingIndicator in Nuxt3, start nor finish methods are exposed or accessible in any way. The state is controlled solely by nuxt hooks "page:start", "page:finish".

Is there a way to control these states manually? Trying to trigger these hooks somehow, seems like it would cause more harm/side effects down the line, than it's worth + I don't think one can tigger these hooks manually anway.

For now I will probably clone the loading component to a vendor folder and ajdust it myself.

**Comment by JoshuaPerk** [comment]
Did you ever find an answer to this, @Gaspadlo?

**Comment by TheAlexLichter** [comment, marked as answer]
You *could* call the page:start/page:finish hook as needed *but* that might have unwanted side effects.

Example:

```ts
const app = useNuxtApp()
app.callHook('page:start')
```

Ideally, the Loading Indicator would expose some kind of API via composable. Ideas/RFC welcome.

**Comment by Gaspadlo** [comment]
@JoshuaPerk Hey. I've not worked on this in months. I slapped this together back then in a hurry. It sure is not pretty, but it gets the job done.
It definitely is not a PR to the main branch material, but if it's any worth to you, feel free to use it.

[AppLoadingIndicator.vue.txt](https://github.com/nuxt/nuxt/files/10855536/AppLoadingIndicator.vue.txt)

Then It can be used like this:
![image](https://user-images.githubusercontent.com/4014669/222003683-529617b3-7b32-408b-9ff1-0564d47a2ed7.png)
![image](https://user-images.githubusercontent.com/4014669/222004048-35077295-ff16-4ee0-9ce4-a762c20b28e8.png)

Result:
![image](https://user-images.githubusercontent.com/4014669/222006428-e0c104a7-ce3b-49e7-a6e1-cf56fa96019d.png)

**Reply by JoshuaPerk** [reply]
This is fantastic! Thank you 🙏🏼

**Reply by acidjazz** [reply]
@JoshuaPerk Can you or someone add a little more detail on how this works? It's not really clear

**Reply by Gaspadlo** [reply]
@acidjazz it mostly just clones nuxt3 built-in loading indicator component and exposes states and methods via useState. It's a bit crudely done, but a proper implementation should not be difficult IMO.
My solution has some "hacky" workarounds for useState composable dropping methods.

**Comment by asugomez** [comment]
IS there any other solution from Nuxt3? still trying to work NuxtLoadingIndicator and useLoadingIndicator
